Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -477,18 +477,25 @@ bool MatrixWorkspace::hasGroupedDetectors() const {
  *    KEY is the DetectorID (pixel ID)
  *    VALUE is the Workspace Index
  *  @param throwIfMultipleDets :: set to true to make the algorithm throw an
- * error
- *         if there is more than one detector for a specific workspace index.
+ * error if there is more than one detector for a specific workspace index.
+ *  @param ignoreIfNoValidDets :: set to true to exclude spectra that do not
```

Comment:
If a spectra had multiple det IDs associated, some valid and some invalid, do we end up dropping all of the IDs associated with the spectra or just return the valid ones?

This might be worth clarifying in the docs string as either:
`exclude spectra (and all associated detector IDs)`
or
`exclude invalid detectors IDs from a spectra`

Reply:
It actually returns all of them, including the invalid ones! I'm really not sure what to do in that case. I doubt it is something that exists in reality. I'll update the doc string though.

Reply:
I'll also add a comment to the test to document the fact that this is not necessarily desired behaviour but is just not important to deal with at the moment